src/js/disability-benefits/526EZ/components/VerifiedReviewPage.jsx

```diff
@@ -0,0 +1,180 @@
+import ProgressButton from '../../../common/components/form-elements/ProgressButton';
+import { setData, setPrivacyAgreement, setEditMode, setSubmission, submitForm, uploadFile } from '../../../common/schemaform/actions';
+
+import SaveFormLink from '../../../common/schemaform/save-in-progress/SaveFormLink';
+import SaveStatus from '../../../common/schemaform/save-in-progress/SaveStatus';
+import { saveAndRedirectToReturnUrl, autoSaveForm } from '../../../common/schemaform/save-in-progress/actions';
+import { toggleLoginModal } from '../../../login/actions';
+import { getNextPagePath, getPreviousPagePath } from '../../../common/schemaform/routing';
+
+import formConfig from '../config/form';
+import ReviewCollapsiblePage from './ReviewCollapsiblePage';
+
+const scroller = Scroll.scroller;
+
+const scrollToTop = () => {
+  scroller.scrollTo('topScrollElement', window.VetsGov.scroll || {
+    duration: 500,
+    delay: 0,
+    smooth: true,
+  });
+};
+
+class VerifiedReviewPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.debouncedAutoSave = _.debounce(1000, this.autoSave);
+  }
+
+  componentDidMount() {
+    if (!this.props.blockScrollOnMount) {
+      scrollToTop();
+      focusElement('h4');
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.route.pageConfig.pageKey !== this.props.route.pageConfig.pageKey ||
+      _.get('params.index', prevProps) !== _.get('params.index', this.props)) {
+      scrollToTop();
+      focusElement('h4');
+    }
+  }
+
+  setData = (...args) => {
+    this.props.setData(...args);
+    this.debouncedAutoSave();
+  }
+
+  autoSave = () => {
+    const { form, user } = this.props;
+    if (user.login.currentlyLoggedIn) {
+      const data = form.data;
+      const { formId, version } = form;
+      const returnUrl = this.props.location.pathname;
+
+      this.props.autoSaveForm(formId, data, version, returnUrl);
+    }
+  }
+
+  goBack = () => {
+    const { form, route: { pageList }, location } = this.props;
+    const path = getPreviousPagePath(pageList, form.data, location.pathname);
+    this.props.router.push(path);
+  }
+
+  goForward = () => {
+    const { form, route, location } = this.props;
+    const path = getNextPagePath(route.pageList, form.data, location.pathname);
+
+    this.props.router.push(path);
+  }
+
+  handleEdit = (pageKey, editing, index = null) => {
+    this.props.setEditMode(pageKey, editing, index);
+  }
+
+  render() {
+    const { form, user, formContext, route } = this.props;
+    const { chapterKey, pageKey } = route.pageConfig;
+    const page = formConfig.chapters[chapterKey].pages[pageKey];
+    const isNotPrefilled = !form.data.prefilled;
+
+    if (isNotPrefilled) return <RoutedSavablePage {...this.props}/>;
+    return (
+      <div>
+        <p>Please review the information we have on file for you. If something doesn’t look right, you can fix it by clicking the Edit button.</p>
```

Comment:
Could this text go in ReviewCollapsiblePage? I'm thinking that eventually we want to get to where we could wrap your custom ReivewCollapsiblePage component in RoutedSavablePage and reduce some of the duplication of SiP code. I don't think that's necessary right now, but I'd want to do that before making this a recommended pattern for forms.

Reply:
Sounds good. Updated.